.github/workflows/pr-checks.yaml

```diff
@@ -97,10 +100,19 @@ jobs:
   test-and-codecov:
     needs: check-changes
     if: needs.check-changes.outputs.changes == 'true'
+    permissions:
```

Comment:
a question here for this.
if https://github.com/sustainable-computing-io/kepler/pull/2345/files#diff-7a6a4578fe04c4fc86b28e2dc9672e48d1a9f721315e540b90bac0ef12263053R6-R8 set global permission for all jobs, why do we need a permission setting at here?